core/src/main/java/org/apache/spark/SparkThrowable.java

```diff
@@ -39,4 +39,7 @@
   // Portable error identifier across SQL engines
   // If null, error class or SQLSTATE is not set
   String getSqlState();
+
+  // True if this error is an internal error.
+  boolean isInternalError();
```

Comment:
java interface can have default implementation now. Shall we use `SparkThrowableHelper.isInternalError(errorClass)` as the default implementation?

Reply:
Oh, good to know. Just pushed a commit with default implementations for `getSqlState` and `isInternalError`.